File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.util.Scanner;

class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int[] coin = {500,100,50,10,5,1};
        int money = 1000 - sc.nextInt();
        int result = 0;
        for(int i = 0 ; i < coin.length ; i++){
            result += money / coin[i];
            money %= coin[i];
        }
        System.out.println(result);
    }
}

File: 백준/Silver/14916. 거스름돈/거스름돈.py
n = int(input())
result = 0
if n < 5:
    if n % 2 == 0 : result = n //2
    else: result = -1
elif n % 5 == 0:
    result = n // 5
elif (n % 5) % 2 == 0:
    result += n // 5
    n = n - (5 * result)
    result += n // 2
else:
    result += (n // 5) - 1
    n = n - (5 * result)
    result += n//2

print(result)

